support/Pipelines/EccentricityControl/InitialOrbitalParameters.py

```diff
@@ -0,0 +1,194 @@
+    The result can be used to start an eccentricity control procedure to tune
+    the initial orbital parameters further.
+
+    Currently only zero eccentricity (circular orbits) are supported, and the
+    implementation uses functions from SpEC's ZeroEccParamsFromPN.py. This
+    should be generalized.
+
+    Arguments:
+      mass_ratio: Defined as q = M_A / M_B >= 1.
+      dimensionless_spin_a: Dimensionless spin of the larger black hole, chi_A.
+      dimensionless_spin_b: Dimensionless spin of the smaller black hole, chi_B.
+      eccentricity: Eccentricity of the orbit. Specify together with _one_ of
+        the other orbital parameters. Currently only an eccentricity of 0 is
+        supported (circular orbit).
+      mean_anomaly_fraction: Mean anomaly of the orbit divided by 2 pi, so it is
+        a number between 0 and 1. The value 0 corresponds to the pericenter of
+        the orbit (closest approach), the value 0.5 corresponds to the apocenter
+        of the orbit (farthest distance), and the value 1 corresponds to the
+        pericenter again. Currently this is unused because only an eccentricity
+        of 0 is supported.
+      separation: Coordinate separation D of the black holes.
+      orbital_angular_velocity: Omega_0.
+      radial_expansion_velocity: adot_0.
+      num_orbits: Number of orbits until merger.
+      time_to_merger: Time to merger.
+
+    Returns: Tuple of the initial separation 'D_0', orbital angular velocity
+      'Omega_0', and radial expansion velocity 'adot_0'.
+    """
+    dimensionless_spin_a = np.asarray(dimensionless_spin_a)
+    dimensionless_spin_b = np.asarray(dimensionless_spin_b)
+    if eccentricity is not None and eccentricity != 0.0:
+        assert mean_anomaly_fraction is not None, (
+            "If you specify a nonzero 'eccentricity' you must also specify a"
+            " 'mean_anomaly_fraction'."
+        )
+    if eccentricity is None:
+        assert (
+            separation is not None
+            and orbital_angular_velocity is not None
+            and radial_expansion_velocity is not None
+        ), (
+            "Specify all orbital parameters 'separation',"
+            " 'orbital_angular_velocity', and 'radial_expansion_velocity', or"
+            " specify an 'eccentricity' plus one orbital parameter."
+        )
+        return separation, orbital_angular_velocity, radial_expansion_velocity
+
+    # The functions from SpEC currently work only for zero eccentricity. We will
+    # need to generalize this for eccentric orbits.
+    assert eccentricity == 0.0, (
+        "Initial orbital parameters can currently only be computed for zero"
+        " eccentricity."
+    )
+    assert radial_expansion_velocity is None, (
+        "Can't use the 'radial_expansion_velocity' to compute orbital"
+        " parameters. Remove it and choose another orbital parameter."
+    )
+    assert (
+        (separation is not None)
+        ^ (orbital_angular_velocity is not None)
```

Comment:
I assume this is `xor`. Does python not allow us to type `xor`? That is easier to read, I find.

Reply:
Not in the language. There's `import operator; operator.xor(a, b)`.

Reply:
Okay, thanks!